drivers/timer/arcv2_timer0.c

```diff
@@ -272,3 +341,17 @@ void sys_clock_disable(void)
 
 	irq_disable(IRQ_TIMER0);
 }
+
+
+#ifdef CONFIG_SMP
+void smp_timer_init(void)
+{
+	/* ensure slave core's internal timer doesn't work */
+	timer0_control_register_set(0);
+	timer0_count_register_set(0);
+	timer0_limit_register_set(0);
```

Comment:
Same here no point of setting `count` & `limit` registers as it won't stop the timer/counter. Just set `control.IE` bit to 0.